libparsec/crates/platform_mountpoint/src/windows/mod.rs

```diff
@@ -0,0 +1,480 @@
+        let fc = file_context.lock().expect("Mutex is poisoned");
+
+        if let Context::File(file) = fc.deref() {
+            self.fd_flush(file.fd);
+        }
+
+        Ok(())
+    }
+
+    fn set_basic_info(
+        &self,
+        _file_context: Self::FileContext,
+        _file_attributes: FileAttributes,
+        _creation_time: u64,
+        _last_access_time: u64,
+        _last_write_time: u64,
+        _change_time: u64,
+    ) -> Result<(), NTSTATUS> {
+        Err(STATUS_NOT_IMPLEMENTED)
+    }
+
+    fn set_file_size(
+        &self,
+        file_context: Self::FileContext,
+        new_size: u64,
+        set_allocation_size: bool,
+    ) -> Result<(), NTSTATUS> {
+        let fc = file_context.lock().expect("Mutex is poisoned");
+
+        if let Context::File(file) = fc.deref() {
+            self.fd_resize(file.fd, new_size, set_allocation_size)?;
+        }
+
+        Ok(())
+    }
+
+    fn rename(
+        &self,
+        _file_context: Self::FileContext,
+        file_name: &U16CStr,
+        new_file_name: &U16CStr,
+        replace_if_exists: bool,
+    ) -> Result<(), NTSTATUS> {
+        let path = PathBuf::from(file_name.to_os_string());
+        let new_path = PathBuf::from(new_file_name.to_os_string());
+        self.entry_rename(&path, &new_path, replace_if_exists)?;
+        Ok(())
+    }
+
+    fn get_security(
+        &self,
+        _file_context: Self::FileContext,
+    ) -> Result<PSecurityDescriptor, NTSTATUS> {
+        Ok(SD.as_ptr())
+    }
+
+    fn set_delete(
+        &self,
+        _file_context: Self::FileContext,
+        file_name: &U16CStr,
+        _delete_file: bool,
+    ) -> Result<(), NTSTATUS> {
+        let path = PathBuf::from(file_name.to_os_string());
+        let stat = self.entry_info(&path)?;
+
+        if path == Path::new("/") {
+            return Err(STATUS_RESOURCEMANAGER_READ_ONLY);
+        } else if !stat.children.is_empty() {
+            return Err(STATUS_DIRECTORY_NOT_EMPTY);
+        }
+
+        Ok(())
+    }
+}
+
+pub fn mount(mountpoint: &Path) -> FileSystem<MountpointManager> {
+    let mountpoint =
+        U16CString::from_os_str(mountpoint.as_os_str()).expect("Unreachable, valid OsStr");
+    let mut volume_params = VolumeParams::default();
+
+    volume_params
+        .set_sector_size(512)
+        .set_sectors_per_allocation_unit(1)
+        .set_volume_creation_time(filetime_now())
+        .set_volume_serial_number(0)
+        .set_file_info_timeout(1000)
+        .set_case_sensitive_search(true)
+        .set_case_preserved_names(true)
+        .set_unicode_on_disk(true)
+        .set_persistent_acls(true)
+        .set_post_cleanup_when_modified_only(true)
+        .set_file_system_name(&mountpoint)
+        .set_prefix(u16cstr!(""));
+
+    let params = Params {
+        volume_params,
+        ..Default::default()
+    };
+
+    FileSystem::new(params, Some(&mountpoint), MountpointManager::default())
+        .expect("Failed to init FileSystem ")
```

Comment:
The idea in platform crates is to return a `anyhow::Error` when something wrong occurred and the caller can't realistically be expecting it (e.g. in platform storage we return a `anyhow::Error` if the sqlite db fails)